### PR TITLE
RuntimeManager and Runtime cleanup

### PR DESCRIPTION
**Pull request by tgrospic**
## Overview

This PR is not changing any functionality but it's cleanup of Runtime changes done for parallel execution of deploys required for block-merge.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/engine/Setup.scala [hunk 1]
```diff
@@ -53,18 +52,16 @@ object Setup {
...
 
-    val (runtime, replayRuntime) = runtimes
     val (exporter, importer) = {
       (historyRepo.exporter.unsafeRunSync, historyRepo.importer.unsafeRunSync)
     }
     implicit val rspaceStateManager = RSpaceStateManagerImpl(exporter, importer)
 
     implicit val runtimeManager =
-      RuntimeManager.fromRuntimes(runtime, replayRuntime, historyRepo).unsafeRunSync(scheduler)
+      RuntimeManager[Task](rspace, replay, historyRepo).unsafeRunSync(scheduler)
```
Comment (nzpr) [inline casper/src/test/scala/coop/rchain/casper/engine/Setup.scala:64]: Are we going to leave it this way? Or the goal to not create manager having initial runtimes passed as input is still on the way?

**Comment by nzpr** [comment]
@tgrospic there is conflict with base branch here, just fyi

File: casper/src/test/scala/coop/rchain/casper/engine/Setup.scala (hunk 1 above)
Comment (tgrospic) [reply casper/src/test/scala/coop/rchain/casper/engine/Setup.scala:64]: Creating RM from runtimes does not make sense. Existing code is just using runtimes to get RSpace and RSpaceReplay which is exactly what now RM constructor accepts.

**Review by nzpr — APPROVED** [review]
LGTM

**Review by zsluedem — APPROVED** [review]
LGTM

**Comment by tgrospic** [comment]
bors merge

**Comment by tgrospic** [comment]
bors merge

**Pull request merged**